interpreter/text/parser.mly

```diff
@@ -347,9 +367,12 @@ expr1 :  /* Sugar */
     { fun c -> let c' = $2 c [] in let ts, es = $3 c' in [], block ts es }
   | LOOP labeling_opt block
     { fun c -> let c' = $2 c [] in let ts, es = $3 c' in [], loop ts es }
-  | IF labeling_opt value_type_list if_
+  | IF labeling_opt if_
     { fun c -> let c' = $2 c [] in
-      let es, es1, es2 = $4 c c' in es, if_ $3 es1 es2 }
+      let es, es1, es2 = $3 c c' in es, if_ [] es1 es2 }
+  | IF labeling_opt LPAR TYPE value_type_list RPAR if_
```

Comment:
`type` isn't the right keyword here, but I wanted to propose changing the syntax of block types anyway, in order to be future compatible with generalised block types. With that, result types need to be enclosed in `(result ...)`, analogous to functions. That might solve your problem as well.